cal-itp/benefits
# Feat: handle OAuth errors

Closes #2019 

* Implements a new error screen in the `oauth` app
* Handles errors on creation/registration of the OAuth client
* Handles errors for all OAuth client method calls
* Sends a new analytics event `oauth error` in all cases
  * `operation` property indicates the specific client call: `init`, `authorize_redirect`, `authorize_access_token`, and `load_server_metadata`
  * `message` property describes the exception
* Captures an exception in Sentry in all cases

## How to test

1. Checkout the branch locally
2. Run `bin/init.sh` to get the PO file updates
3. Launch the app with `F5`
4. Select an agency or enter via their slug URL
5. Manually enter the URL in the browser: `/oauth/error`: see the error page
6. Attempt a Login.gov flow for the agency (without configuring the `AuthProvider`): see the error page
7. Misconfigure an `EligibilityVerifier` to not specify an `AuthProvider` _and_ without an Eligibility API base URL or form class and attempt to sign in with this verifier selected: see the error page
